plugins/modules/ndo_l3out_node_group_policy.py

```diff
@@ -0,0 +1,344 @@
+
+    mso_template = MSOTemplate(mso, "l3out", template)
+    mso_template.validate_template("l3out")
+
+    l3out_object = mso_template.get_l3out_object(name=l3out, fail_module=True)
+    l3out_node_group = mso_template.get_l3out_node_group(name, l3out_object.details)
+
+    if name and l3out_node_group:
+        mso.existing = mso.previous = copy.deepcopy(l3out_node_group.details)  # Query a specific object
+    elif l3out_node_group:
+        mso.existing = l3out_node_group  # Query all objects
+
+    if state != "query":
+        node_group_policy_path = "/l3outTemplate/l3outs/{0}/nodeGroups/{1}".format(l3out_object.index, l3out_node_group.index if l3out_node_group else "-")
+
+    ops = []
+
+    if state == "present":
+        l3out_node_routing_policy_object = None
+        if node_routing_policy:
+            l3out_node_routing_policy_objects = mso.query_objs(
+                generate_api_endpoint(
+                    "templates/objects", **{"type": "l3OutNodePolGroup", "tenant-id": mso_template.template_summary.get("tenantId"), "include-common": "true"}
+                )
+            )
+            l3out_node_routing_policy_object = mso_template.get_object_by_key_value_pairs(
+                "L3Out Node Routing Policy", l3out_node_routing_policy_objects, [KVPair("name", node_routing_policy)], True
+            )
+
+        if mso.existing:
+            proposed_payload = copy.deepcopy(mso.existing)
+            mso_values = dict()
+            mso_values_remove = list()
+
+            if proposed_payload.get("bfdMultiHop", {}).get("key", {}).get("ref"):
+                proposed_payload["bfdMultiHop"]["key"].pop("ref", None)
+                mso.existing["bfdMultiHop"]["key"].pop("ref", None)
+
+            mso_values["description"] = description
+
+            if node_routing_policy == "" and mso.existing.get(
+                "nodeRoutingPolicyRef"
+            ):  # Clear the node routing policy when node_routing_policy is empty string
+                mso_values_remove.append("nodeRoutingPolicyRef")
+            else:
+                mso_values["nodeRoutingPolicyRef"] = l3out_node_routing_policy_object.details.get("uuid") if l3out_node_routing_policy_object else None
+
+            if bfd:
+                if check_if_all_elements_are_none(list(bfd.values())) and proposed_payload.get("bfdMultiHop"):
+                    mso_values_remove.append("bfdMultiHop")
+                elif not check_if_all_elements_are_none(list(bfd.values())):
+                    if not proposed_payload.get("bfdMultiHop"):
+                        mso_values["bfdMultiHop"] = dict()
+
+                    mso_values[("bfdMultiHop", "authEnabled")] = ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP.get(bfd.get("auth"))
+                    mso_values[("bfdMultiHop", "keyID")] = bfd.get("key_id")
+
+                    if bfd.get("key") is not None:
+                        mso_values[("bfdMultiHop", "key")] = dict(value=bfd.get("key"))
```

Comment:
Think we can do the foloowing
```suggestion
                    mso_values[("bfdMultiHop", "key", "value")] = bfd.get("key")
```

Reply:
Yes, we can directly update the value when the key is present. Otherwise, we should use a dict object to update the value.